### DISCUSSION TITLE
Connector / API for Power Platform and Logic Apps scenarios

### DISCUSSION THREAD

**Discussion by pettersh** [Puzzlepart/did#1092] [Feature request]
The current did API (GraphQL) isn't optimized for integration with external systems.

I suggest an additional REST based API following Open API spec to ease integrations with (preferrable supported by Azure AD):

- Power Automate / Logic Apps
- Power Apps (Canvas + Model)
- Dynamics 365 (CRM, ERP)
- SharePoint Framework
- Xledger, Agresso, Visma, etc...

Edit: It doesn't have to be a REST API, but it should be easy to consume and interact with did without exessive knowledge about the internal structure of did. Maybe it can be achieved with documentation and example integrations?

**Comment by olemp** [comment]
@pettersh @okms @damsleth I'm not very keen on maintaining both a REST API and our current GraphQL API to be honest.

For Power Platform a custom connector can be used, and I'm guessing Dynamics can consume GraphQL somehow. SPFx can easily use GraphQL.

**Reply by pettersh** [reply]
The link you shared with me @olemp https://ashiqf.com/2021/04/25/what-is-graphql-and-how-to-consume-a-graphql-query-based-api-in-power-automate/

**Reply by olemp** [reply]
I see there's some articles on automatically generate REST API endpoints from GraphQL. Would be nice if we could keep one "API codebase"; but still offer REST API endpoints. https://www.moesif.com/blog/technical/graphql/Generating-Rest-APIs-from-GraphQl-Schemas/

**Comment by olemp** [comment]
I think we should rewind and think how do we want to integrate with did? Teams app? Bot? Viva Connections?

Then we could look into what we need from the GraphQL API , and how we do it in the desired "platform" (SPFx, Teams, whatever).
